lib/services/hook.js

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015, EMC, Inc.
+
+'use strict';
+
```

Comment:
Overall, hook is better not to be a service, in my thinking, the difference between Hook, Messenger and Services.Messenger is as following:
(1) For the term "Messenger", it is a kind of channel that supports publish/subscribe message. The channel can be AMQP or Hook.
(2) The current di module `Messenger` is better to rename to "Messenger.AMQP", it is a class that publish/subscribe message via AMQP channel.
(3) The Hook can be casted to a messenger, it is a class that publish/subscribe message via Web URL. But in my vision of Hook, it can do more than a messenger. So Hook is not equal to a messenger but it can behave as a messenger.
(4) The "Services.Messenger" is to manage all message channels, that means all AMQP messenger and Hook Messenger instances should be initialized and registered in Services.Messenger.

Add @sunnyqianzhang @iceiilin @anhou